### DISCUSSION TITLE
How to get started with mill from the command line

### DISCUSSION THREAD

**Discussion by DieBauer** [com-lihaoyi/mill#3375] [Q&A]
Hi, I want to start using Mill, however I couldn't find how to get the latest version and start using that from the CLI. 

I see the website dives immediately into examples and why you want to use it, but nowhere a one-liner like `brew install mill` or `sdk install mill` or `curl ...`. 

How should I get this tool?

**Comment by lefou** [comment]
The best way to get started is to download the [bootstrap script](https://mill-build.org/mill/0.11.11/Java_Installation_IDE_Support.html#_bootstrap_scripts). I recommend to use [lefou/millw](https://github.com/lefou/millw), since it also runs on Windows and automatically scans for updates if there is no explicit version configured. It will download and run Mill for you. You can either keep it local to your project or install it into your personal `bin` folder.

So in short, either:

```sh
curl -L https://github.com/com-lihaoyi/mill/releases/download/0.11.11/0.11.11 > mill && chmod +x mill
```

or 

```sh
curl -L https://raw.githubusercontent.com/lefou/millw/0.4.12/millw > mill && chmod +x mill
```

If you are on Windows, you can use the batch script

```bat
curl -L https://raw.githubusercontent.com/lefou/millw/0.4.11/millw.bat > mill.bat
```

or the power shell script

```ps1
curl -L https://raw.githubusercontent.com/lefou/millw/0.4.11/millw.ps1 > mill.ps
```

If you find the documentation not clear enough or have some idea how to improve it, feel free to open a pull request.
